OTC-811: Validation of insuree number fix

olewandowski1 opened the pull request:
[OTC-811](https://openimis.atlassian.net/browse/OTC-811)

In scope of this ticket, I corrected the validation of the insuree number. Now, when it is invalid, there is no possibility to save the form (save button is disabled).
Moreover, the warning message has been changed from "_Invalid Insuree Number_" to "_Insurance number already in use_".

Related BE: [CORE PR](https://github.com/openimis/openimis-be-core_py/pull/148) and [INSUREE PR](https://github.com/openimis/openimis-be-insuree_py/pull/47)

[OTC-811]: https://openimis.atlassian.net/browse/OTC-811?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

dborowiecki commented on src/translations/en.json:
```diff
@@ -222,7 +222,7 @@
   "insuree.addInsuree.alert.title": "Add insuree to family warning(s):",
   "insuree.addInsuree.alert.message": "However, insuree can still be added to the family.",
-  "insuree.InsureeNumberInput.error": "Invalid Insuree Number",
+  "insuree.InsureeNumberInput.error": "Insurance number already in use",
```
Please remember about updating lokalise after changing labels.

olewandowski1 replied:
It is already updated.

dborowiecki reviewed (changes requested):
I think we need to reiterate on insuree number validation.

olewandowski1 commented on src/pickers/InsureeNumberInput.js:
```diff
@@ -1,19 +1,25 @@
+  } = props;
   const modulesManager = useModulesManager();
-  const numberMaxLength = modulesManager.getConf("fe-insuree", "insureeForm.chfIdMaxLength", 12)
+  const numberMaxLength = modulesManager.getConf("fe-insuree", "insureeForm.chfIdMaxLength", 12);
```
It is configured in FE directly. This length restriction does not allow to enter more than 12 chars. I get your point, so I will proceed working on this if BE will be ready.

olewandowski1 commented:
@dborowiecki We changed the way how the information about invalid insuree number is shown. Now, the displayed information is taken straight from the payload as you suggested. Let me know if this solution meets your expectation. I'm curious about the translation now, how should it look now? Btw. I removed the unused key from Lokalise.

olewandowski1 commented:
@dborowiecki I changed the way of displaying those error messages. The reason of changes was that the messages must translated and the previous approach made it impossible. Could you please take a look and confirm that this approach meets your expectations now. If so, I will add those phrases to the Lokalise.

dragos-dobre commented:
Some considerations here: 
- take the message error from BE but currently the code doesn't provide same details as the actual message sent back from BE (i.e. expected length) 
- if BE messages are displayed directly, then translation need to be handled on BE side
- limits in allow insuree number length defined in BE (Damian's comment)

dborowiecki commented:
@dragos-dobre I agree, we should make message more informative by adding length. I would prefer to do this adjustment on frontend and treat the message from the backend only as an additional piece of information about what the error code referees to. 
I think it be more consistent with the other validations and frontend translations, as if I'm not mistaken, the only place where translated backend strings are used is in the mutation result bar on the right site. Let me know what do you think about this approach.

delcroip commented:
@olewandowski1 please check the comments

olewandowski1 commented:
@delcroip I saw those comments, but at the moment it is done as was requested by @dborowiecki. I'm using codes to choose which type of issue message should be displayed. @dragos-dobre had some considerations, so I asked Damian for his statement on it. And I can see that Dragos did not reply yet, so that's the reason why I'm still waiting for.

If you have some time, please take a look on those comments, maybe do you have some idea how should we deal with those translations? Should I keep them on FE or should it be done on the BE?

dragos-dobre commented:
The validation is not done in save button so right bar solution doesn't work. be error should be displayed under input.

olewandowski1 commented:
@dragos-dobre @dborowiecki So I reverted last commit and removed the FE translation. At the momment I'm using the messages returned from a BE.
